scenedetect/cli/context.py

```diff
@@ -396,6 +396,46 @@ def handle_detect_threshold(
             ))
 
         self.options_processed = options_processed_orig
+    
+    def handle_detect_hash(
+        self,
+        threshold: Optional[float],
+        min_scene_len: Optional[str],
+        hash_size: Optional[int],
+        highfreq_factor: Optional[int]
+    ):
+        """Handle detect-hash command options."""
+        self._check_input_open()
+        options_processed_orig = self.options_processed
+        self.options_processed = False
+
+        if self.drop_short_scenes:
+            min_scene_len = 0
+        else:
+            if min_scene_len is None:
+                if self.config.is_default("detect-hash", "min-scene-len"):
+                    min_scene_len = self.min_scene_len.frame_num
+                else:
+                    min_scene_len = self.config.get_value("detect-hash", "min-scene-len")
+            min_scene_len = parse_timecode(min_scene_len, self.video_stream.frame_rate).frame_num
```

Comment:
Probably time to refactor this into a helper function (e.g. `get_min_scene_len("detect-hash")` or something) and use that here and in the other detectors.

Reply:
I implemented a version of this in 688d67e as a private function of the `CliContext` class (`_get_min_scene_len(command)`). I tested it with all the detectors (and found a typo from an earlier commit of mine in the process) and it seems to work for all of them. Let me know if you think this should be done another way.